api/src/main/java/org/openmrs/api/TagService.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api;
+
+import org.openmrs.OpenmrsObject;
+import org.openmrs.TagConstants;
+import org.openmrs.annotation.Authorized;
+import org.openmrs.api.OpenmrsService;
+import org.openmrs.Tag;
+
+import java.util.List;
+
+/**
+ * The main service of this module, which is exposed for other modules. See
+ * moduleApplicationContext.xml on how it is wired up.
+ */
+
+public interface TagService extends OpenmrsService {
+	
+	/**
+	 * Returns a tag by uuid. It can be called by any authenticated user. It is fetched in read only
+	 * transaction.
+	 * 
+	 * @param uuid
+	 * @return
+	 */
+	@Authorized(TagConstants.MANAGE_TAGS)
```

Comment:
You will need to add all the privileges used by your module to be added in the config.xml file